### PR TITLE
Created new workflow for CI.

### PR DESCRIPTION
**Pull request by kianapaz**
Runs tests and linter using flake8

### CODE REVIEW

File: .github/workflows/continuous_integration.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
+        python-version: ["3.8", "3.9", "3.10"]
```
Comment (NolanVH) [inline .github/workflows/continuous_integration.yml:19]: The SRS doc specifies that we will only be using a linux environment. We could probably remove the MacOS and Windows build matrices. Additionally, the PyPy setup file requires that python 3.9 or later is used.

**Review by NolanVH — CHANGES_REQUESTED** [review]
Other than the minor changes as suggested in my comment this all looks good to me. Nice work @kianapaz!

File: .github/workflows/continuous_integration.yml (hunk 1 above)
Comment (kianapaz) [reply .github/workflows/continuous_integration.yml:19]: Sounds good thank you!

Comment (NolanVH) [reply .github/workflows/continuous_integration.yml]: Resolved in d76010fa0722d00c4db5803b5aa6afcd52c18f2e

**Review by NolanVH — DISMISSED** [review]

File: .github/workflows/continuous_integration.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Test with pytest
+      run: |
+        pip install .
+        cd tests
+        pytest
```
Comment (Ckoelewyn) [inline .github/workflows/continuous_integration.yml:43]: Should always end a file with a newline

Comment (kianapaz) [reply .github/workflows/continuous_integration.yml]: fixed!

**Review by Ckoelewyn — APPROVED** [review]
Looks great

**Review by strdavis — APPROVED** [review]
Looks great thank you!

**Pull request merged**